### PR TITLE
WIP: CI: Automated Linux builds

### PR DESCRIPTION
**Pull request by hexagon6**
* [x] add .travis.yml
* [x] adjust settings to match for `BusterLabs/Partyshare` at `deploy` key: 
  * [x] change branch to master
  * [x] change repo to `Busterlabs/Partyshare`
  * [x] review if tagged commits should trigger the pipeline `tags: true`
  * [x] @bhstahl: generate travis api_key via travis utility by running in the repo folder: `$ gem install travis; travis setup releases`

snippet from .travis.yml for help
```
deploy:
  provider: releases
  api_key:
    secure: <your key>
  file: dist/Partyshare.deb
  skip_cleanup: true
  on:
    repo: hexagon6/Partyshare
    branch: ci/upstream-build
tags: true
```

### CODE REVIEW

**Comment by bhstahl** [comment]
@hexagon6, I'm excited to get started on a OSX build pipeline, so I just put together https://github.com/BusterLabs/Partyshare/pull/48, which added a travis configuration file for us to work concurrently on our respective targets

**Comment by bhstahl** [comment]
With #49, I setup an OSX build & release pipeline using travis, and used it to build & upload release [v1.2.3](https://github.com/BusterLabs/Partyshare/releases/tag/v1.2.3). 

When you're ready, lets get #42 merged and this target added to release `v1.2.4` with a linux app! 🍾

File: .travis.yml [hunk 1]
```diff
@@ -24,6 +35,7 @@ deploy:
     - dist/latest-mac.yml
```
Comment (bhstahl) [inline .travis.yml]: When you run `yarn build_electron` does that produce `dist/github/latest-linux.json` & `dist/latest-linux.yml` for you? If so, we may want to include them for auto-update support

File: .travis.yml [hunk 2]
```diff
@@ -24,6 +38,7 @@ deploy:
...
     - dist/partyshare.busterlabs.xyz-*-mac.dmg
     - dist/partyshare.busterlabs.xyz-*-mac.zip
+    - dist/Partyshare.deb
```
Comment (bhstahl) [inline .travis.yml]: I believe the linux target _should_ follow the [`artifactName` scheme](https://github.com/BusterLabs/Partyshare/blob/b7c059d983eb8bb756fac994d7b687b0244ea2cd/package.json#L53), producing `partyshare.busterlabs.xyz-1.2.7-linux.tar.gz` and `partyshare.busterlabs.xyz-1.2.7-linux.deb` for the latest release?

File: .travis.yml (hunk 1 above)
Comment (hexagon6) [reply .travis.yml]: No,
I only get this:

```bash
$ ls dist/
linux-unpacked  partyshare.busterlabs.xyz-1.2.7-linux.deb  partyshare.busterlabs.xyz-1.2.7-linux.tar.gz
```

I'll check how electron deals with Linux builds. Have to change the file reference too :-)

File: .travis.yml (hunk 2 above)
Comment (hexagon6) [reply .travis.yml]: @bhstahl true, just noticed, see my comment above

File: .travis.yml (hunk 1 above)
Comment (bhstahl) [reply .travis.yml]: Ah, no worries. Auto-update might not be supported on linux, I'll check.

Comment (hexagon6) [reply .travis.yml]: ok, you're right. unfortunately. I noticed my build crashed when trying to update. Can/should we disable asar for linux builds?

Comment (bhstahl) [reply .travis.yml]: Hmm, it looks like we should be creating a `latest-linux.*` file [according to the docs](https://www.electron.build/auto-update#file-generated-and-uploaded-in-addition)

Comment (hexagon6) [reply .travis.yml]: locally I don't get one. maybe do a `ls -l` at `before_deploy` key in `.travis.yml`, so we can see what is actually there.

Comment (hexagon6) [reply .travis.yml]: Regarding the "crash":

$ ipfs daemon; partyshare.busterlabs.xyz

My log output: https://gist.github.com/hexagon6/14303b59b269877eafcd384b7dc17909

It crashes with `ipfs version 0.3.10`, probably I should update.

**Comment by hexagon6** [comment]
@bhstahl could you trigger a draft build so we see the contents of dist/ after building on travis?

File: .travis.yml (hunk 1 above)
Comment (hexagon6) [reply .travis.yml]: Nope, even with version 0.4.13 I get this Error:

```
_getConfig Error: non-zero exit code 1

            while running: /opt/Partyshare/resources/app.asar.unpacked/node_modules/go-ipfs-dep/go-ipfs/ipfs config show


            Error: api not running

    at Stream.listeners.done.once (/opt/Partyshare/resources/app.asar/node_modules/ipfsd-ctl/src/exec.js:39:11)
    at Stream.f (/opt/Partyshare/resources/app.asar/node_modules/once/once.js:25:25)
    at emitTwo (events.js:125:13)
    at Stream.emit (events.js:213:7)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
```

**Comment by bhstahl** [comment]
Hmm, it's not letting me trigger a build since the code is in your fork. Let's merge this and work off master builds & PR's. Thanks for your help!

**Comment by bhstahl** [comment]
Alright, looks like we didn't get the auto-update files, but we do have the `deb` and `tar.gz` in the [draft](https://github.com/BusterLabs/Partyshare/releases/tag/untagged-391bca757dfe1268e8b5).

**Pull request merged**